aptos-move/aptos-vm-types/src/change_set.rs

```diff
@@ -370,13 +554,35 @@ impl VMChangeSet {
     pub(crate) fn extend_resource_write_set(
         &mut self,
         patched_resource_writes: impl Iterator<Item = (StateKey, WriteOp)>,
-        finalized_group_writes: impl Iterator<Item = (StateKey, WriteOp)>,
-    ) {
-        self.resource_write_set.extend(
-            patched_resource_writes
-                .chain(finalized_group_writes)
-                .map(|(k, v)| (k, (v, None))),
-        );
+    ) -> Result<(), PanicError> {
+        for (k, new_write) in patched_resource_writes {
+            match self.resource_write_set.entry(k) {
```

Comment:
completely optional re:style, but in the case when Vacant leads to error, might be worth to consider mut entry = ...get_mut(k).ok_or_else(code_invariant_error..)?;
and then the occupied(mut o/entry) logic - less indentation.